tuf/keys.py

```diff
@@ -1016,8 +1053,66 @@ def format_rsakey_from_pem(pem):
   # Ensure the PEM string has a valid header and footer.  Although a simple
   # validation of 'pem' is performed here, a fully valid PEM string is
   # needed to later successfully verify signatures.
-  pem_header = '-----BEGIN PUBLIC KEY-----'
-  pem_footer = '-----END PUBLIC KEY-----'
+  public_pem = extract_pem(pem) 
+
+  # Begin building the RSA key dictionary. 
+  rsakey_dict = {}
+  keytype = 'rsa'
+  public = public_pem 
+
+  # Generate the keyid of the RSA key.  'key_value' corresponds to the
+  # 'keyval' entry of the 'RSAKEY_SCHEMA' dictionary.  The private key
+  # information is not included in the generation of the 'keyid' identifier.
+  key_value = {'public': public,
+               'private': ''}
+  keyid = _get_keyid(keytype, key_value)
+
+  rsakey_dict['keytype'] = keytype
+  rsakey_dict['keyid'] = keyid
+  rsakey_dict['keyval'] = key_value
+
+  return rsakey_dict
+
+
+
+
+
+def extract_pem(pem, private_pem=False):
```

Comment:
Unclear if resolved:

Vlad, your intention with [the comments added here in PR #311](https://github.com/theupdateframework/tuf/pull/311/files#diff-a3be85d1a9209356a591a86fedcad4a5R1097) is to indicate that (thorough) validation here is not necessary because we can trust that pyca/cryptography and PyCrypto will perform any necessary validation, and any tinkering will result in a pem that is unusable?